stablehlo/dialect/TypeInference.cpp

```diff
@@ -3366,6 +3368,80 @@ LogicalResult verifyCollectivePermuteOp(
   return success();
 }
 
+LogicalResult verifyCompositeOp(std::optional<Location> loc, Operation* op,
+                                StringRef name, StringRef decomposition) {
+  // composite_c1
+  SmallVector<StringRef> parts;
+  name.split(parts, '.');
+  if (
+      // Name must have at least 2 non-empty parts separated by periods.
+      parts.size() < 2 ||
```

Comment:
Let's make this its own function: `LogicalResult verifyNamespacedOpName(...)`.
That way we can have a nice code comment for what a valid name is here as well.

Alternative: Any reason not to use the regex to test this as opposed to hand-parsing?